### DISCUSSION TITLE
System adjusting buttons failed

### DISCUSSION THREAD

**Discussion by ShawnDominicDDD** [kyleneideck/BackgroundMusic#538] [Q&A]
Hi developer,
Having launched the app, the volume setting buttons on my keyboard fail to work.
Any ideas about that?

**Comment by kyleneideck** [comment]
Thanks for reporting it. I'm not getting the same issue, but you could have a look in the debug logs and see if they tell you anything useful: https://github.com/kyleneideck/BackgroundMusic/wiki/Getting-Debug-Logs

You'll only need to look at the logs for `Background Music.app` (BGMApp) for this, not the Background Music driver (BGMDriver), so that makes it a bit easier. Feel free to post your logs here, but I probably won't have time to do more than have a quick look through them.

The code that's not working for you is probably https://github.com/kyleneideck/BackgroundMusic/blob/master/BGMApp/BGMApp/BGMDeviceControlSync.cpp which calls functions in https://github.com/kyleneideck/BackgroundMusic/blob/master/BGMApp/BGMApp/BGMAudioDevice.cpp

What output device are you using (i.e. the hardware)? I'd guess that code might not work properly for all devices. Or maybe you're using a different version of macOS or something. Feel free to open a bug report for this if you want.
